ui/raidboss/data/02-arr/trial/titan-nm.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+[{
+  zoneId: ZoneId.TheNavel,
+  timelineFile: 'titan-nm.txt',
+  triggers: [
+    {
+      id: 'TitanNm Tumult',
+      netRegex: NetRegexes.ability({ id: '282', source: 'Titan', capture: false }),
+      condition: Conditions.caresAboutAOE(),
+      suppressSeconds: 2,
+      response: Responses.aoe(),
+    },
+    {
+      // Gaol callout for both yourself and others
+      id: 'TitanNm Gaols',
+      netRegex: NetRegexes.gainsEffect({ effectId: '124' }),
+      alertText: function(data, matches) {
+        if (matches.target !== data.me) {
+          return {
+            en: 'Break Gaol on ' + data.ShortName(matches.target),
+          };
+        }
+        return {
+          en: 'Gaol on YOU',
+        };
+      },
+    },
+    {
+      // Covers both tanks and non tanks with appropriate text
+      // Needs suppression for occasions in phase 4 where rock busters
+      // get pushed close together by auto attacks to avoid double triggers.
+      id: 'TitanNm Rock Buster',
+      netRegex: NetRegexes.ability({ id: '281', source: 'Titan' }),

```

Comment:
I think this needs to be a timeline trigger.  (Maybe this is what @JLGarber is saying below as well?)  This ability has no castbar and so you can't do a `startsUsing`.  If you use `ability` here, this will go off after the ability has already happened.  So, if you want to do it ahead of time, you need to use a timeline trigger.  (Maybe this is what you were asking about on discord a while back?)

Reply:
Would I just move all instances of Rock Buster a couple seconds back in the timeline, so it goes off early enough, or how would I do this properly?

Reply:
You can use `beforeSeconds` with a timeline trigger.  There's examples in both of the other Titan files.

Reply:
Quisquous is correct, this is a timeline trigger, sorry I didn't make that clearer! What you want to do is set it up something like this:

```javascript
timelineTriggers: [
    {
      id: 'TitanNm Rock Buster',
      regex: /Rock Buster/,
      beforeSeconds: 7,
      condition: function(data) {
        return data.role == 'healer' || data.role == 'tank';
      },
      response: Responses.tankBuster(),
    },
],
```

(Shamelessly stolen from [Titan EX](https://github.com/quisquous/cactbot/blob/60e3b23d44ff538f8ab248845b502365f38765de/ui/raidboss/data/02-arr/trial/titan-ex.js#L6).)

As you can see, we use a normal `Regex` property, because we're not looking for a log line, we're looking for a timeline match. The `beforeSeconds` property tells the timeline playback that "okay, you have the timeline file loaded, when the timer reaches 7 seconds before any line that contains Rock Buster, activate the trigger".